vec3.h

```diff
@@ -0,0 +1,120 @@
+//code by Ameya Vikrama Singh
+#include<iostream>
+#include<cmath>
+class vec3{
+    public:
+        float x,y,z;
+
+        vec3(){
+            x=y=z=0;
+        }
+
+        //Cartesian coordinate Constructor
+        vec3(float x,float y,float z){
+            this->x=x;
+            this->y=y;
+            this->z=z;
+        }
+        
+        //assignment operator
+        void operator=(vec3 s){
+            x=s.x;
+            y=s.y;
+            z=s.z;
+        }
+
+        //norm function
+        float norm(){
+            return sqrt(x*x+y*y+z*z);
+        }
+
+        //vector addition
+        vec3 operator+(vec3 s){
+            vec3 sum;
+            sum.x=s.x+x;
+            sum.y=s.y+y;
+            sum.z=s.z+z;
+            return sum;
+        }
+
+        //vector subtraction
+        vec3 operator-(vec3 s){
+            vec3 diff;
+            diff.x=x-s.x;
+            diff.y=y-s.y;
+            diff.z=z-s.z;
+            return diff;
+        }
+
+        //addition+assignemnt
+        void operator+=(vec3 s){
+            x+=s.x;
+            y+=s.y;
+            z+=s.z;
+        }
+
+        //subtraction+assignemnt
+        void operator-=(vec3 s){
+            x-=s.x;
+            y-=s.y;
+            z-=s.z;
+        }
+
+        //scalar post-multiplication
+        vec3 operator*(float s){
+            vec3 prod;
```

Comment:
Why declare `prod` here and then assign to it in the next three lines? You can just do `return vec3(s * x, s * y, s * z);`, or even better, `return {s * x, s * y, s * z};`. The latter is a [C++17 feature](https://stackoverflow.com/a/35244051/12591388), and the modern way to write it.